protocol/x/affiliates/keeper/keeper.go

```diff
@@ -176,46 +178,47 @@ func (k Keeper) GetTierForAffiliate(
 	if err != nil {
 		return 0, 0, err
 	}
-	numTiers := uint32(len(affiliateTiers.GetTiers()))
+	tiers := affiliateTiers.GetTiers()
+	numTiers := uint32(len(tiers))
 	maxTierLevel := numTiers - 1
 	currentTier := uint32(0)
 	referredVolume, err := k.GetReferredVolume(ctx, affiliateAddr)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	for index, tier := range affiliateTiers.GetTiers() {
-		if referredVolume.Int64() >= int64(tier.ReqReferredVolume) {
+	for index, tier := range tiers {
+		if referredVolume.Cmp(lib.BigU(tier.ReqReferredVolumeQuoteQuantums)) >= 0 {
 			// safe to do as tier cannot be negative
 			currentTier = uint32(index)
 		}
 	}
 
 	if currentTier == maxTierLevel {
-		return currentTier, affiliateTiers.GetTiers()[currentTier].TakerFeeSharePpm, nil
+		return currentTier, tiers[currentTier].TakerFeeSharePpm, nil
 	}
 
 	numCoinsStaked := k.statsKeeper.GetStakedAmount(ctx, affiliateAddr)
 	for i := currentTier + 1; i < numTiers; i++ {
-		expMultiplier := new(big.Int).Exp(
-			big.NewInt(10),
-			big.NewInt(-lib.BaseDenomExponent),
-			nil,
-		)
+		expMultiplier, _ := lib.BigPow10(lib.BaseDenomExponent)
```

Comment:
We are consciously using `BaseDenomExponent` which is a negative value and ignoring `inverse` boolean and relying on `lib.BigPow10(` implementation behavior that returns the same value for both positive and negative exponents. 

For robustness and readability, better to use `-lib.BaseDenomExponent` instead?

Reply:
Fair point. I assumed thats how it was used. I'll fix